docs/codegen/generate-file.md

```diff
@@ -64,3 +64,24 @@ or even the project root, which then generates the file for multiple stacks.
 There is no overriding or merging behavior for `generate_file` blocks.
 Blocks defined at different levels with the same label aren't allowed, resulting
 in failure of the overall code generation process.
+
+
+## Conditional Code Generation
+
+Conditional code generation is achieved by the use of the `condition` attribute.
+The `condition` attribute should always evaluate to a boolean. If `condition` 
+is true, the code will be generated for the given stack. If `condition` is false,
+the code will **not** be generated for the given stack.
+
```

Comment:
maybe makes sense to state that if `condition` is not provided, then the code is always generated.. but not sure, it seems clear enough.